src/main/java/io/fabric8/maven/docker/access/hc/DockerAccessWithHcClient.java

```diff
@@ -704,19 +698,26 @@ private void doPushImage(String url, Map<String, String> header, HcChunkedRespon
         }
     }
 
-    private String tagTemporaryImage(ImageName name, String registry) throws DockerAccessException {
+    private TemporaryImageHandler tagTemporaryImage(ImageName name, String registry) throws DockerAccessException {
         String targetImage = name.getFullName(registry);
-        if (!name.hasRegistry() && registry != null) {
-            if (hasImage(targetImage)) {
-                throw new DockerAccessException(
-                    String.format("Cannot temporarily tag %s with %s because target image already exists. " +
-                                  "Please remove this and retry.",
-                                  name.getFullName(), targetImage));
-            }
-            tag(name.getFullName(), targetImage, false);
-            return targetImage;
+        if (name.hasRegistry() || registry == null) {
+            return interruptingError ->
+                log.info("Temporary image tag skipped. Target image '%s' already has registry set or no registry is available");
```

Comment:
You should also check if the `interruptingError` is non-null and log it accordingly. The same is applicable at line 717, where another lambda implements `TemporaryImageHandler`. You could probably make a second class implementation that accepts printed text on execution, but if error is present, it also appends its message.

Reply:
I've made changes that will make `interruptingError` to be thrown even in case of handlers that only log message. It is done by `TemporaryImageHandler#handle(DockerAccessException)`